### PR TITLE
test: Code Coverage: View-Team Reports Component #2

### PR DESCRIPTION
**Pull request by jayfyle**
## Coverage at 99%:

view_team_report_99

## Branch Coverage Improved:

view_team_report_99_2

### CODE REVIEW

**Review by yksolanki9 — CHANGES_REQUESTED** [review]
Organize the assertions inside ionViewWillEnter as they are executed in the code.

**Review by yksolanki9 — CHANGES_REQUESTED** [review]

File: src/app/fyle/view-team-report/view-team-report.page.spec.ts [hunk 1]
```diff
@@ -218,6 +219,261 @@ describe('ViewTeamReportPage', () => {
...
+      component.erpt$
+        .pipe(
+          finalize(() => {
+            expect(loaderService.hideLoader).toHaveBeenCalledTimes(1);
+          })
+        )
+        .subscribe((res) => {
+          expect(res).toEqual(expectedAllReports[0]);
+        });
+
```
[the remark this answers is not in the document]
Comment (jayfyle) [reply src/app/fyle/view-team-report/view-team-report.page.spec.ts:352]: This is the final subscribe for the block, can't access the intermediate assignment for `this.erpts$`, the previous matcher was incorrect. Have fixed it now.

File: src/app/fyle/view-team-report/view-team-report.page.spec.ts [hunk 2]
```diff
@@ -329,15 +338,17 @@ describe('ViewTeamReportPage', () => {
...
-          })
-        )
-        .subscribe((res) => {
-          expect(res).toEqual(expectedAllReports[0]);
-        });
+      component.erpt$.subscribe((res) => {
+        expect(res).toEqual(expectedAllReports[0]);
+      });
+
+      expect(component.systemComments).toEqual(userCommentsData);
```
Comment (yksolanki9) [inline src/app/fyle/view-team-report/view-team-report.page.spec.ts:345]: Add comments other than policy and system for this test case

File: src/app/fyle/view-team-report/view-team-report.page.spec.ts [hunk 3]
```diff
@@ -329,15 +338,17 @@ describe('ViewTeamReportPage', () => {
...
+
+      expect(component.objectType).toEqual('reports');
+
+      expect(component.systemEstatuses).toEqual(updateReponseWithFlattenedEStatus);
+
+      expect(component.userComments).toEqual(userCommentsData);
```
Comment (yksolanki9) [inline src/app/fyle/view-team-report/view-team-report.page.spec.ts:351]: This is not correct. System comments and user comments cannot be the same. Please don't modify the api response to suit the tests.

File: src/app/fyle/view-team-report/view-team-report.page.spec.ts [hunk 4]
```diff
@@ -218,6 +219,261 @@ describe('ViewTeamReportPage', () => {
...
+          finalize(() => {
+            expect(loaderService.hideLoader).toHaveBeenCalledTimes(1);
+          })
+        )
+        .subscribe((res) => {
+          expect(res).toEqual(expectedAllReports[0]);
+        });
+
+      expect(reportService.getExports).toHaveBeenCalledOnceWith(activatedRoute.snapshot.params.id);
+      expect(reportService.getApproversByReportId).toHaveBeenCalledOnceWith(activatedRoute.snapshot.params.id);
```
[the remark this answers is not in the document]
Comment (yksolanki9) [reply src/app/fyle/view-team-report/view-team-report.page.spec.ts:354]: Missing assertion for sharedWith

**Review by yksolanki9 — APPROVED** [review]

**Pull request merged**